[feat] update find_top_k to filter database

InayaY3 opened the pull request:
## What's new in this PR
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
I created a new function in Supabase that builds off of Carolyn's `find_top_k` function, now adding functionality for filtering by other factors (gender, age, veteran status, offense, and state). This has also been updated in `query.tsx`, which now makes a call to `find_top_k_filtered` and takes in the needed arguments.


## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

**Edit: also check out the supabase for the new function and to see what the tables being used look like!

To test via useEffects, create a new folder (called "test" or whatever) within app. In this folder, create a page.tsx file. You can copy the code from below into the file, and mess around with the filter values (gender, age, etc.) to see what returns. Otherwise, run the app normally and inspect/check the console.log. Keep in mind this is only using ~11 dummy rows from the adoptee_vector table.
``` 
'use client';

import { useEffect } from 'react';
import { fetchTopK } from '../../actions/queries/query';

export default function TestTopKPage() {
 useEffect(() => {
   async function test() {
     const embedding = Array.from(
       { length: 384 },
       () => Math.random() * 2 - 1,
     );
     const k_value = 3;
     const gender = 'Female';
     const age = 70;
     const veteran_status = 'Unknown';
     const offense = null;
     const state = 'Wyoming';
     const data = await fetchTopK(
       embedding,
       k_value,
       gender,
       age,
       veteran_status,
       offense,
       state,
     );
     console.log('Top K results:', data);
     console.log(
       'For the vals of: ',
       gender,
       age,
       veteran_status,
       offense,
       state,
     );
   }
   test();
 }, []);

 return <div>Check the browser console duhhh</div>;
}
```

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
Based on the NPO's feedback, we will use a hierarchy to handle cases where there are less than k matches being returned in the array.


## Relevant links
### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'


### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
building off of [carolyn's old sprint](https://github.com/calblueprint/adopt-an-inmate/pull/22)


CC: @ethan-tam33

ethan-tam33 commented on package.json:
```diff
@@ -53,6 +53,7 @@
     "prettier": "^3.3.3",
     "prettier-plugin-tailwindcss": "^0.6.14",
     "tailwindcss": "^4.1.13",
+    "ts-node": "^10.9.2",
```
Nice! is there a reason you installed ts-node though? seems like we're not using this rn?

InayaY3 replied:
oops forgot to delete, was back when i was trying to figure out how i should test

ethan-tam33 commented on actions/queries/query.ts:
```diff
@@ -14,11 +14,24 @@ export async function fetchAllRows() {
+) {
   const supabase = await getSupabaseServerClient();
-  const { data, error } = await supabase.rpc('find_top_k', {
+  const { data, error } = await supabase.rpc('find_top_k_filtered', {
```
nice! can you alter the format to take data from the `adoptee_vector_test` table instead, and then get all of its columns that aren't already mentioned in `fetchTopK`? Should be a quick fix!

ethan-tam33 commented:
Oh and also do you know why this PR has a change to pnpm-lock.yaml? I don't think it should've changed right?

InayaY3 commented:
> Oh and also do you know why this PR has a change to pnpm-lock.yaml? I don't think it should've changed right?

I think it's from when I added tsx. I removed it from everywhere else but I looked into it and can't remove it from pnpm-lock.yaml without altering things we actually need even further. It's showing up as a peer dependency of vite-related stuff but it's not actually installed or used so shouldn't affect anything?

ethan-tam33 commented:
Just tested it, and it works though!

ethan-tam33 reviewed (changes requested):
can you try doing `pnpm install` again? I don't think the `pnpm-lock.yaml` file should be changed... But your filtering works!
